course/enrollment.py

```diff
@@ -76,6 +76,49 @@
 
 # }}}
 
+MESSAGE_ENROLLMENT_SENT_TEXT = _(
+    "Enrollment request sent. You will receive notifcation "
+    "by email once your request has been acted upon.")
+MESSAGE_ENROLL_REQUEST_PENDING_TEXT = _(
+    "Your enrollment request is pending. You will be "
+    "notified once it has been acted upon.")
+MESSAGE_ENROLL_DENIED_NOT_ALLOWED_TEXT = _(
+    "Your enrollment request had been denied. Enrollment is not allowed.")
+MESSAGE_ENROLL_DROPPED_NOT_ALLOWED_TEXT = _(
+    "You had been dropped from the course. Re-enrollment is not allowed.")
+MESSAGE_ENROLL_REQUEST_ALREADY_PENDING_TEXT = _(
+    "You have previously sent the enrollment request. "
+    "Re-sending the request is not allowed.")
+MESSAGE_PARTICIPATION_ALREADY_EXIST_TEXT = _(
+    "A participation already exists. Enrollment attempt aborted.")
+MESSAGE_CANNOT_REENROLL_TEXT = _("Already enrolled. Cannot re-enroll.")
+MESSAGE_SUCCESSFULLY_ENROLLED_TEXT = _("Successfully enrolled.")
+MESSAGE_EMAIL_SUFFIX_REQUIRED_PATTERN = _(
+    "Enrollment not allowed. Please use your '%s' email to enroll.")
+MESSAGE_NOT_ACCEPTING_ENROLLMENTS_TEXT = _("Course is not accepting enrollments.")
+MESSAGE_ENROLL_ONLY_ACCEPT_POST_REQUEST_TEXT = _(
+    "Can only enroll using POST request")
+MESSAGE_ENROLLMENT_DENIED_TEXT = _("Successfully denied.")
+MESSAGE_ENROLLMENT_DROPPED_TEXT = _("Successfully dropped.")
+
+MESSAGE_BATCH_PREAPPROVED_RESULT_PATTERN = _(
+    "%(n_created)d preapprovals created, "
+    "%(n_exist)d already existed, "
+    "%(n_requested_approved)d pending requests approved.")
+
+MESSAGE_EMAIL_NOT_CONFIRMED_TEXT = _(
+    "Your email address is not yet confirmed. "
+    "Confirm your email to continue.")
+MESSAGE_PARTICIPATION_CHANGE_SAVED_TEXT = _("Changes saved.")
+
+EMAIL_NEW_ENROLLMENT_REQUEST_TITLE_PATTERN = (
+    string_concat("[%s] ", _("New enrollment request")))
+EMAIL_ENROLLMENT_DECISION_TITLE_PATTERN = (
+    string_concat("[%s] ", _("Your enrollment request")))
+
+VALIDATION_ERROR_USER_NOT_CONFIRMED = _(
+    "This user has not confirmed his/her email.")
```

Comment:
What made you decide to define these? I'm asking because it seems easier and more direct to just keep messages where they're used. And it's easier to tell what a message says by looking at what it actually says vs. a symbolic name representing what it says.

Reply:
Sorry, I think it's easier to test whether the correct messages are displayed in the page requested, in unittest. I will revert that because your opinion is more reasonable than mine.

Reply:
Thanks for being accommodating! The tests will highlight when they're out of sync, so it should be an easy fix.